table/tables/partition_test.go

```diff
@@ -518,3 +518,58 @@ func (ts *testSuite) TestHashPartitionInsertValue(c *C) {
 	result := tk.MustQuery("SELECT * FROM t4 WHERE a = 1")
 	result.Check(testkit.Rows("\x01 1"))
 }
+
+func (ts *testSuite) TestIssue31629(c *C) {
+	tk := testkit.NewTestKitWithInit(c, ts.store)
+	tk.MustExec("create database Issue31629")
+	defer tk.MustExec("drop database Issue31629")
+	tk.MustExec("use Issue31629")
+	// Test following partition types:
```

Comment:
LIST | RANGE COLUMNS partition on 4.0 is treated as non-partitioned table?

Reply:
Yes, I removed the tests from the unit-test but forgot to update the comment. It was noticed by the check for `SHOW WARNINGS` after the create table too.